src/qutip_cupy/__init__.py

```diff
@@ -5,6 +5,13 @@
 # to an official optional dependency
 import warnings
 
+try:
+    import cupy
+except ModuleNotFoundError:
+    raise RuntimeError('''qutip_cupy requires cupy to be installed, please install
+                         cupy by following the instructions at https://docs.cupy.dev/en/stable/install.html''')
+del cupy
```

Comment:
what would deleting cupy imply?

Reply:
`del cupy` just removes the `.cupy` attribute from this module, which we want to do because otherwise people might do `qutip_cupy.cupy` which we don't want them to do.

As an alternative, we could do `__import__("cupy")` which wouldn't set the attribute. Let's do that instead! :D